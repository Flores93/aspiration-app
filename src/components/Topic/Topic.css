.topic-container {
  border: 1px solid rgb(207, 213, 236);
  border-radius: 10px;
  padding: 10px;
  background-color: rgb(223, 230, 255);
  max-width: 300px;
}

.no-topic {
  border: 1px solid rgb(207, 213, 236);
  background-color: rgb(209, 209, 209);
}

.topic-title {
  margin-bottom: 10px;
  color: rgb(104, 104, 104);
}

.child-topics-container {
  display: flex;
  flex-wrap: wrap;
}
