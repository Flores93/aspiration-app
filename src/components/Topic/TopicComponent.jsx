import ChildTopic from "../ChildTopic";
import "./Topic.css";
const Topic = ({ edge, changeTopic }) => {
  const { resourcePath, stargazers, repositoryTopics } = edge.node;

  const existsRelatedTopics = repositoryTopics.nodes.length;

  return (
    <div className={`topic-container ${!existsRelatedTopics && "no-topic"}`}>
      <div className="topic-title">
        <h3>
          {resourcePath}
          {" - "}
          <i className="fa fa-star star-stargazers" />
          <span style={{ color: "grey" }}>{stargazers.totalCount} </span>
        </h3>
      </div>
      <div>
        {existsRelatedTopics ? (
          <>
            {" "}
            <p className="topic-title">Related Topics:</p>
            <div className="child-topics-container">
              {repositoryTopics.nodes.map((node, index) => (
                <ChildTopic node={node} key={index} changeTopic={changeTopic} />
              ))}
            </div>
          </>
        ) : (
          <b>This path it doesn't have any related topic. :(</b>
        )}
      </div>
    </div>
  );
};

export default Topic;
