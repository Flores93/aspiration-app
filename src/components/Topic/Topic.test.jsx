import { mount } from "enzyme";
import Topic from "./TopicComponent";

describe("Topic Component Tests", () => {
  const changeTopicMock = jest.fn();

  const edgeTopicsEmptyMock = {
    node: {
      repositoryTopics: {
        nodes: [],
        totalCount: 0,
      },
      resourcePath: "/dummy/path",
      stargazers: { totalCount: 3842 },
    },
  };

  const edgeMock = {
    ...edgeTopicsEmptyMock,
    node: {
      ...edgeTopicsEmptyMock.node,
      repositoryTopics: {
        nodes: [
          {
            topic: {
              name: "faketopic1",
              stargazerCount: 121,
              relatedTopics: [
                { name: "fakeRelatedTopic-1", _typename: "Topic" },
                { name: "fakeRelatedTopic-2", _typename: "Topic" },
              ],
            },
          },
          {
            topic: {
              name: "faketopic2",
              stargazerCount: 33,
              relatedTopics: [
                { name: "fakeRelatedTopic-3", _typename: "Topic" },
                { name: "fakeRelatedTopic-4", _typename: "Topic" },
              ],
            },
          },
          {
            topic: {
              name: "faketopic3",
              stargazerCount: 923,
              relatedTopics: [
                { name: "fakeRelatedTopic-5", _typename: "Topic" },
                { name: "fakeRelatedTopic-6", _typename: "Topic" },
              ],
            },
          },
          {
            topic: {
              name: "faketopic4",
              stargazerCount: 77,
              relatedTopics: [
                { name: "fakeRelatedTopic-7", _typename: "Topic" },
                { name: "fakeRelatedTopic-8", _typename: "Topic" },
              ],
            },
          },
        ],
        totalCount: 4,
      },
    },
  };

  test("Should show the component correctly", () => {
    const wrapper = mount(
      <Topic edge={edgeMock} changeTopic={changeTopicMock} />
    );

    const topicTitle = wrapper.find("h3").text();
    const relatedTopicsLabel = wrapper.find("p").text();
    const childTopicsNumber = wrapper.find("div.child-topic");

    expect(wrapper).toMatchSnapshot();

    expect(typeof topicTitle).toBe("string");
    expect(topicTitle).toContain(edgeTopicsEmptyMock.node.resourcePath);
    expect(relatedTopicsLabel).toBe("Related Topics:");
    expect(childTopicsNumber).toHaveLength(
      edgeMock.node.repositoryTopics.totalCount
    );
  });

  test("Should show the message of empty related topics", () => {
    const wrapper = mount(
      <Topic edge={edgeTopicsEmptyMock} changeTopic={changeTopicMock} />
    );

    const noTopicsMessage = wrapper.find("b").text();
    const existsEmptyContainerClass = wrapper.find(".no-topic").exists();

    expect(noTopicsMessage).toEqual(
      "This path it doesn't have any related topic. :("
    );
    expect(existsEmptyContainerClass).toBe(true);
  });
});
