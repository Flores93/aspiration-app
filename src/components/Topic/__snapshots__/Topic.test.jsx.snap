// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic Component Tests Should show the component correctly 1`] = `
<div
  className="topic-container false"
>
  <div
    className="topic-title"
  >
    <h3>
      /dummy/path
       - 
      <i
        className="fa fa-star star-stargazers"
      />
      <span
        style={
          Object {
            "color": "grey",
          }
        }
      >
        3842
         
      </span>
    </h3>
  </div>
  <div>
     
    <p
      className="topic-title"
    >
      Related Topics:
    </p>
    <div
      className="child-topics-container"
    >
      <div
        className="child-topic"
        onClick={[Function]}
      >
        <b>
          faketopic1
          : 
        </b>
        <span>
           
          121
           
          <i
            className="fa fa-star mr-4 star-stargazers"
          />
        </span>
      </div>
      <div
        className="child-topic"
        onClick={[Function]}
      >
        <b>
          faketopic2
          : 
        </b>
        <span>
           
          33
           
          <i
            className="fa fa-star mr-4 star-stargazers"
          />
        </span>
      </div>
      <div
        className="child-topic"
        onClick={[Function]}
      >
        <b>
          faketopic3
          : 
        </b>
        <span>
           
          923
           
          <i
            className="fa fa-star mr-4 star-stargazers"
          />
        </span>
      </div>
      <div
        className="child-topic"
        onClick={[Function]}
      >
        <b>
          faketopic4
          : 
        </b>
        <span>
           
          77
           
          <i
            className="fa fa-star mr-4 star-stargazers"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
